src/Icicle/Avalanche/Simp.hs

```diff
@@ -88,7 +88,7 @@ simpFlattened a_fresh p
    -- Pull Let statements out of blocks. This just allows thresher to remove more duplicates
    >>= lift . nestBlocks   a_fresh
    -- Thresh to remove duplicate expressions
-   >>= thresher     a_fresh
+   >>= thresherNoAlpha     a_fresh
```

Comment:
luckily not: because lets are non-recursive, the only need for alpha checks are before flattening, where lambdas are used